src/transform.ts

```diff
@@ -117,7 +117,16 @@ function process() {
             break;
           }
           case TemplateKeyType.IF:
-            checkIf = template[key];
+            checkIf = template[key].filter(element => typeof element === 'string');
+            try {
+              if (template[key]
+                .filter(cb => typeof cb === 'function')
+                .map(cb => cb(options.$, options.$1))
+                .some(val => !val)) return;
+            } catch (err) {
+              if (!err.message?.includes('Cannot read property')) console.warn(`Warning: ${err.message}`);
+              return;
+            }
```

Comment:
❓ Can you try this for me? I think it's equivalent as $if is a valid template (actually a subset, with this it will support more cases eg: $if:['keyInResult', '$.keyInInput', ($)=> $.anotherKeyInInput !== undefined] ).

```suggestion
            checkIf = transform(template[key], {...options, dropValues:[]}); // keeping also undefined values etc.
            if(checkIf.some(val => !val)) return;
            checkIf = checkIf.filter(element => typeof element === 'string');
```

Reply:
after some testing the above will work, but line 66 will need to change from
`return compact(template.map(t => transform(t, options)));` to
`return compact(template.map(t => transform(t, options)), [],  options.dropValues);`
Since `compact` function default dropValues is actually [undefined].

Secondly, your example `$if:['keyInResult', '$.keyInInput', ($)=> $.anotherKeyInInput !== undefined] )` seemed to work  fine, however, if the `$.keyInInput` resolves to a string, then the produced checkIf will become
`['keyInResult', 'the string value of $keyInInput', true]`
This means, that after the filtering, it will try to search if the key with name `the string value of $keyInInput` has a non undefined value, which will obviously have. 

There is no straightforward way unfortunately to check each of the remaining strings inside the checkif whether they are plain values or actual keys.

Reply:
Thanks for checking it @johnvlach! The change in L66 is possibly needed anyway to respect the top dropValues anyway.
You are right about this ['keyInResult', 'the string value of $keyInInput', true]. 

Both version are ok with me. Can you try to merge now? I approved the ci run

Reply:
> Thanks for checking it @johnvlach! The change in L66 is possibly needed anyway to respect the top dropValues anyway.
> You are right about this ['keyInResult', 'the string value of $keyInInput', true].
> 
> Both version are ok with me. Can you try to merge now? I approved the ci run

unfortunately I cannot merge it. No merge button :(